repositories/slot_repository.go

```diff
@@ -0,0 +1,22 @@
+package repositories
+
+import (
+	"airport-app-backend/models"
+)
+
+type ISlotRepository interface {
+	RetrieveAllSlots(int, string) ([]models.Slot, error)
+}
+
+func (sr *ServiceRepository) RetrieveAllSlots(page int, status string) ([]models.Slot, error) {
+	var slots []models.Slot
+	
+	query := sr.db.Offset(page * DEFAULT_PAGE_SIZE).Limit(DEFAULT_PAGE_SIZE)
+	if status != "ALL" {
```

Comment:
Dont use a string - we don't need to support more than 2 possible values, which can be done easier via a boolean